target_quickbooks/mapper.py

```diff
@@ -6,6 +6,42 @@
 from datetime import datetime
 
 
+class EntityNotFoundException(Exception):
+    pass
+
+
+def lookup_entity(record, id_field, name_field, entity, entity_list: dict):
+    entity_id = None
+    if record.get(id_field) or record.get(name_field):
+        if record.get(id_field) in [v["Id"] for v in entity_list.values()]:
+            entity_id = record.get(id_field)
+        elif record.get(name_field):
+            entity_id = entity_list.get(record.get(name_field), {}).get("Id")
+        if not entity_id:
+            raise EntityNotFoundException(f"Could not find {entity} in Quickbooks matching Id= {id_field} or Name={name_field}")
```

Comment:
It would be good to check with Bren if we should fail in all entities that are not found or only on required fields, in other targets we would fail if the field is required and if not just log and send the data without it